plugins/inputs/mesos/mesos.go

```diff
@@ -828,6 +829,20 @@ func generateTaggedField(parts []string) TaggedField {
 	return tf
 }
 
+// decodeFrameworkName returns a framework name from its encoded representation, as it would be found in a Mesos
+// metrics response. If decoding fails, it logs a warning and returns the encoded name.
+func decodeFrameworkName(encoded string) string {
+	// https://mesos.apache.org/documentation/latest/monitoring/#frameworks
+	// > The framework name is percent-encoded before creating these metrics; the actual name can be recovered by
+	// > percent-decoding.
+	decoded, err := url.QueryUnescape(encoded)
```

Comment:
I checked, and QueryUnescape does decode percent encoding, so this will work fine: https://play.golang.org/p/APwTGI_3g23